### DISCUSSION TITLE
Invalid recipient troubleshooting

### DISCUSSION THREAD

**Discussion by hallamjeff** [YoRyan/mailrise#4] [Q&A] [answered]
I'm running mailrise in Docker (using docker-compose) and can successfully send a test apprise notice when inside the container (docker exec -it) using the same `matrixs://[...]` line from my mailrise config. However, when trying to send mail to the docker container from another machine using:

`echo "Subject: hello" | sendmail matrix@<docker host>`

I consistently receive:

`sendmail: RCPT TO:<matrix@<docker host>> (550 'matrix@<docker host>' is not a valid mailrise recipient)`

My docker-compose is fairly simple and I'm having trouble figuring out why mailrise doesn't seen `matrix@...` as a valid recipient:

```configs:
  matrix:
    urls:
      - matrixs://selfhosted_bot:<password>@matrix.org/<room id>:matrix.org
```

Any help would be appreciated.

**Comment by YoRyan** [comment, marked as answer]
So Mailrise is actually not intended to handle email for an entire host like this. (It is assumed that if you have a functional email setup with an open relay, then you would just send regular email.) All addresses _must_ end in `@mailrise.xyz`, and you should be using `<docker host>` as your global SMTP server. If you want to retain the ability to send email to other destinations, then you can configure your system to [only forward](https://serverfault.com/a/257640) `mailrise.xyz` addresses to Mailrise.

**Reply by hallamjeff** [reply]
Ok, I see. I've adjusted my test message and now getting a malformed room error in the docker logs so I'll chase that down. I was using the internal room id which starts with "!" but will try an alias instead.

**Reply by hallamjeff** [reply]
Found the malformed room error. Thanks for the help, I had assumed that "mailrise.xyz" was used as an example address when first setting it up.

**Reply by YoRyan** [reply]
No problem! That's a fair point, and I could certainly clarify it in the documentation.
